plugins/search-backend-node/src/IndexBuilder.ts

```diff
@@ -92,37 +92,42 @@ export class IndexBuilder {
   }
 
   /**
-   * Starts the process of executing collators and decorators and building the
-   * search index.
-   *
-   * TODO: But like with coordination, timing, error handling, and what have you.
+   * Compiles collators and decorators into tasks, which are added to a
+   * scheduler returned to the caller.
    */
-  async build() {
-    return Promise.all(
-      Object.keys(this.collators).map(async type => {
+  async build(): Promise<{ scheduler: Scheduler }> {
+    const scheduler = new Scheduler({ logger: this.logger });
+
+    Object.keys(this.collators).map(type => {
```

Comment:
Would argue that `.forEach` is nicer now that the return value isn't used anymore. Also got `Object.entries` at your disposal if you wanna use that :grin: